fe/fe-core/src/main/java/com/starrocks/connector/CatalogConnectorMetadata.java

```diff
@@ -184,7 +184,7 @@ public List<RemoteFileInfo> getRemoteFileInfos(Table table, List<String> partiti
     }
 
     @Override
-    public List<RemoteFileInfo> getRemotePartitions(Table table, List<String> partitionNames) {
+    public List<PartitionInfo> getRemotePartitions(Table table, List<String> partitionNames) {
```

Comment:
why you modify this interface in this patch? I didn't find any place to call this patch

Reply:
This code. Partition Info(last modified time) is not from HMS, but from HDFS. 

```Java
// Used to check whether data in hdfs has changed.
// Hdfs has api to check whether a directory has changed after update,
// So just check the directory of partition, not list all files under parition directory.
public class DirectoryBasedUpdateArbitrator extends TableUpdateArbitrator {
    @Override
    public Map<String, Optional<HivePartitionDataInfo>> getPartitionDataInfos() {
        Map<String, Optional<HivePartitionDataInfo>> partitionDataInfos = Maps.newHashMap();
        List<String> partitionNameToFetch = partitionNames;
        if (partitionLimit >= 0 && partitionLimit < partitionNames.size()) {
            partitionNameToFetch = partitionNames.subList(partitionNames.size() - partitionLimit, partitionNames.size());
        }
        List<PartitionInfo> partitions =
                GlobalStateMgr.getCurrentState().getMetadataMgr().getRemotePartitions(table, partitionNameToFetch);
        for (int i = 0; i < partitionNameToFetch.size(); i++) {
            PartitionInfo partitionInfo = partitions.get(i);
            HivePartitionDataInfo hivePartitionDataInfo = new HivePartitionDataInfo(partitionInfo.getModifiedTime(), 1);
            partitionDataInfos.put(partitionNameToFetch.get(i), Optional.of(hivePartitionDataInfo));
        }
        return partitionDataInfos;
    }
}

```